src/wovnio/wovnphp/Store.php

```diff
@@ -79,6 +87,10 @@ private function defaultSettings() {
      * @return array The new settings of the user
      */
     public function updateSettings($vals=array()) {
+      if (isset($vals['default_lang']) === false) {
+        return array();
```

Comment:
Why reset all setting if the default language is not present?
Is it new from this PR?

Reply:
This is existing behavior,
https://github.com/WOVNio/WOVN.php/pull/28/files#diff-949fcf6c77696a804a40a883c4af4753L26

before:
If there isn't a file, use array().
If there is no "default_lang", error happens

Reply:
> If there is no "default_lang", error happens

I don't see that in the previous code?

Reply:
at `$vals['default_lang']`.
php raise error when the value is not in dictionary(array)

Reply:
Then, can we merge setting with default ones as done at `https://github.com/WOVNio/WOVN.php/blob/master/src/wovnio/wovnphp/Store.php#L32`.
Or we could just prevent the error at `updateSettings()`.

That way we avoid error and don't reset settings.

Reply:
What is the problem of resetting setting when a setting is invalid?
I just make almost the same behavior with now.

If the problem is about current behavior, this is out of the scope of this PR.

Reply:
I disagree with resetting every option if one is missing.
I agree with setting default values for missing "required" options, but not resetting all.

Reply:
My question is " the problem is about current behavior" or not.
If your disagreement is about current behavior, this should be discussed in different place.
Because we didn't discuss anything about how to handle invalid configuration (for me, raising error is better but there are other options).

Reply:
If you want error, then why is this change needed?
I don't see code raising explicit error. If not, then this change is not needed and may lead to error later. So, can't we just discard this change and let the line `$vals['default_lang'] = Lang::getCode($vals['default_lang']);` raise error as now?

Reply:
ok done